tests/foreman/longrun/test_capsule_loadbalancer.py

```diff
@@ -16,74 +16,198 @@
 
 :Upstream: No
 """
+import queue
+import threading
+from pathlib import PurePath
+from time import sleep
+
 import pytest
+from broker import Broker
 
+from robottelo.cli.capsule import Capsule as ContentCapsule
+from robottelo.config import settings
+from robottelo.helpers import InstallerCommand
+from robottelo.hosts import Capsule
+from robottelo.hosts import ContentHost
 
-@pytest.mark.stubbed
-@pytest.mark.tier1
-def initial_setup():
-    """Initial setup
 
-    :id: 6269a87a-e774-11ea-9cb6-4ceb42ab8dbc
-
-    :Steps:
-        1. Add the manifest into the satellite.
-        2. Make sure the subscriptions for required repositories are enabled.
-        3. Sync the repositories.
-        4. Create lifecycle environment, content view and activation key for capsules and client.
-        5. Install and register 2 capsules.
-            5.1 Add the respective lifecycle environment to capsule and sync.
-            5.2 Setup the puppetCA and puppet agent amongst the capsules accordingly.
-        6. Create a host group with default values.
-        7. Install loadbalancer(haproxy)
-            7.1 Algorithms as follows:
-                a) :HTTPS - Source
-                b) :HTTP - RoundRobin
-                c) :AMQP - RoundRobin
-                d) :Puppet - RoundRobin
-                e) :RHSM - RoundRobin
-                f) :SCAP - RoundRobin
-            7.2 Enable logging for haproxy (if disabled)
-
-    :expectedresults: Initial setup complete
+@pytest.fixture(scope='module')
+def get_hosts_from_broker():
+    """Get hosts from broker.
+       Using threading to reduce the checkout time.
+    :return: Capsule and haproxy host
     """
 
+    def _get_hosts(vm_nick, vm_count):
+        host_class = Capsule if 'cap' in vm_nick else ContentHost
 
-@pytest.mark.stubbed
-@pytest.mark.tier1
-def test_katello_cert_download_manually():
-    """Download katello-ca-consumer-latest.noarch.rpm through LB
+        host = Broker(nick=vm_nick, host_classes={'host': host_class}, _count=vm_count).checkout()
+        que.put({vm_nick: host})
 
-    :id: 9398d59e-e141-11ea-83a6-4ceb42ab8dbc
+    threads = list()
+    que = queue.Queue()
+    results = dict()
+    host_count = [['rhel7', 1], ['cap612', 2]]
+    for host in host_count:
+        threads.append(threading.Thread(target=_get_hosts, args=(host[0], host[1])))
+        sleep(1)
+        threads[-1].start()
+    _ = [t.join() for t in threads]
 
-    :Steps: run `yum localinstall -y \
-            http://loadbalancer.example.com/pub/katello-ca-consumer-latest.noarch.rpm`
+    while not que.empty():
+        results.update(que.get())
 
-    :expectedresults: Katello cert should be installed from any one of the capsules.
+    yield results
 
-    """
-    pass
+    all_hosts = [results['rhel7'], results['cap612'][0], results['cap612'][1]]
```

Comment:
you may need to manually run `teardown` on these hosts. they can become too complex for pickle during checkin

Reply:
Changed the way of checkout of hosts. Going with fixtures now